xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/source/MacroContentSourceReferenceConverter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.macro.source;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Convert an object into a {@link MacroContentSourceReference}.
+ * 
+ * @param <T> the type of the value to convert to {@link MacroContentSourceReference}
+ * @version $Id$
+ * @since 15.1RC1
+ * @since 14.10.5
+ */
+@Role
+@Unstable
+public interface MacroContentSourceReferenceConverter<T>
```

Comment:
Same question regarding `CodeMacroSourceReferenceConverter` vs `MacroContentSourceReferenceConverter` (and a bunch of other classes). Do you think it's working factoring the common logic (would make it easier to add a source param to other macros I guess)

Reply:
Same answer than for [CodeMacroSourceReference](https://github.com/xwiki/xwiki-platform/pull/2064/files#diff-e0831740ac54933e017711686298ab0a0bac4e27f067c3f3e32353a6ea717ff3).

Reply:
Questino answered by another [comment](https://github.com/xwiki/xwiki-rendering/pull/243#discussion_r1087541522).